### DISCUSSION TITLE
Please consider adding input field type classes somewhere

### DISCUSSION THREAD

**Discussion by darylknight** [solspace/craft-freeform#332] [Feature Requests]
Using the Tailwind formatting template (although I think this applies to all templates), there are no hooks in the default formatting template to be able to style certain kinds of inputs. For example, if I want to use CSS to change the layout or general style of a checkbox group, there is nothing in the template I can use to target that kind of field. In the screenshot below, there's a row, a wrapping div and a label, then more wrapping divs around each checkbox, but there's nothing different on this HTML to say, a default plain text input field.

Please consider adding something like `freeform-fieldtype` to a wrapping div of each input type so they can be targeted with CSS or JS.

I know the idea is that the built-in templates are "starting points", but frankly there's too much going on in Freeform to make maintaining a custom template worthwhile. Making minor modifications to the default templates is far more maintainable and less likely to break between updates than having a completely custom written one.

![image](https://user-images.githubusercontent.com/1250017/166243567-1c834bb5-5987-4f78-b06e-3ad062b04ef4.png)

**Comment by kjmartens** [comment]
Thanks for the suggestion @darylknight. I've updated the sample formatting templates in Freeform 4+ to now include this. I'm a little bit wary about adding to the existing ones in Freeform 3.

**Comment by darylknight** [comment]
That makes sense! Thank you.
